File: extract_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

let dataBuffer = fs.readFileSync('Mi carta (1).pdf');

pdf(dataBuffer).then(function(data) {
    fs.writeFileSync('temp_menu.txt', data.text);
    console.log('Success, extracted', data.text.length, 'characters.');
}).catch(function(err) {
    console.error('Error extracting PDF:', err);
});

File: extract_pdf.py
import fitz # PyMuPDF
doc = fitz.open("Mi carta (1).pdf")
text = ""
for page in doc:
    text += page.get_text()
with open("extracted_menu.txt", "w", encoding="utf-8") as f:
    f.write(text)
print("Extracted menu to extracted_menu.txt. Length:", len(text))
